ethereum/spec/src/test/java/tech/pegasys/teku/spec/config/SpecConfigBuilderTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2022 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.config;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import java.util.function.Consumer;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.units.bigints.UInt256;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecFactory;
+
+class SpecConfigBuilderTest {
+
+  @Test
+  public void shouldLoadAltairForkEpoch() {
+    final Spec spec =
+        getSpec(
+            phase0Builder ->
+                phase0Builder.altairBuilder(
+                    altairBuilder -> altairBuilder.altairForkEpoch(UInt64.valueOf(10))));
+
+    assertThat(spec.getGenesisSpec().getConfig().getRawConfig().get("ALTAIR_FORK_EPOCH"))
+        .isEqualTo(UInt64.valueOf(10));
+  }
+
+  @Test
+  public void shouldLoadMergeForkEpoch() {
+    final Spec spec =
+        getSpec(
+            phase0Builder ->
+                phase0Builder.mergeBuilder(
+                    mergeBuilder -> mergeBuilder.mergeForkEpoch(UInt64.valueOf(683921))));
+
+    assertThat(spec.getGenesisSpec().getConfig().getRawConfig().get("MERGE_FORK_EPOCH"))
+        .isEqualTo(UInt64.valueOf(683921));
+  }
+
+  @Test
+  public void shouldLoadTerminalTotalDifficulty() {
+    final Spec spec =
+        getSpec(
+            phase0Builder ->
+                phase0Builder.mergeBuilder(
+                    mergeBuilder ->
+                        mergeBuilder.terminalTotalDifficulty(UInt256.valueOf(12_000_000))));
+
+    assertThat(spec.getGenesisSpec().getConfig().getRawConfig().get("TERMINAL_TOTAL_DIFFICULTY"))
+        .isEqualTo(UInt256.valueOf(12_000_000));
+  }
+
+  @Test
+  public void shouldLoadTerminalBlockHash() {
+    final Spec spec =
+        getSpec(
+            phase0Builder ->
+                phase0Builder.mergeBuilder(
+                    mergeBuilder ->
+                        mergeBuilder.terminalBlockHash(
+                            Bytes32.fromHexString(
+                                "0x0000000000000000000000000000000000000000000000000000922192989272"))));
```

Comment:
can use `Bytes32.random` most likely...

I'd probably store it as a static private at the top of the suite...

Reply:
Actually it might be worth instantiating DataStructureUtil, and it can do all the test data stuff you might like...

```
private final Spec mainnetSpec = TestSpecFactory.createMainnetSpec();
private final DataStructureUtil data = new DataStructureUtil(mainnetSpec);
```

then
```
data.randomBytes32();
data.randomUInt64();
```

Reply:
I've now used `Bytes32.random()`. `TestSpecFactory` does not have a method `createMainnetSpec()`.